onadata/apps/logger/models/version.py

```diff
@@ -0,0 +1,71 @@
+from django.db import models
+
+
+class VersionTreeException(Exception):
+    pass
+
+
+class VersionManager(models.Manager):
+    def get_root_path(self, vt):
+        """Get path from root to given vt"""
+        path = []
+        while vt is not None:
+            path.insert(0, vt)
+            vt = vt.parent
+        return path
+
+    def find_path(self, vt1, vt2):
+        """Find and return path from vt1 to vt2
+        :rtype: tuple of two lists ([path up], [path down])
+        """
+        vt1_root_path = self.get_root_path(vt1)
+        vt2_root_path = self.get_root_path(vt2)
+
+        if vt1_root_path[0] != vt2_root_path[0]:
+            raise VersionTreeException(
+                'Path not found. Passed nodes (vt1 = %s, vt2 = %s) '
+                'are from different version trees' % (str(vt1), str(vt2))
+            )
+        last_common_node = last_common_item(vt1_root_path, vt2_root_path)
+        return (vt1_root_path[last_common_node:][::-1],
+                vt2_root_path[last_common_node + 1:])
+
+
+def last_common_item(xs, ys):
```

Comment:
you should be able to use builtin bisect here: https://docs.python.org/2/library/bisect.html#searching-sorted-lists
what you want should be roughly equivalent to index function

Reply:
after discussion - let's drop bisect, since it only complicates things without significant performance gains

Reply:
Done